app/models/user.rb

```diff
@@ -9,4 +12,7 @@ class User < ApplicationRecord
   validates_integrity_of :profile_photo
   validates_processing_of :profile_photo
 
+  def subscribed?(plan)
+    subscriptions.exists?(plan_id: plan.id)
+  end
```

Comment:
if you only need `plan.id` here then only send id. No need to send the whole plan object.